lib/sinatra/base.rb

```diff
@@ -1057,13 +1057,16 @@ def route_missing
     # Attempt to serve static files from public directory. Throws :halt when
     # a matching file is found, returns nil otherwise.
     def static!(options = {})
-      return if (public_dir = settings.public_folder).nil?
-      path = File.expand_path("#{public_dir}#{URI_INSTANCE.unescape(request.path_info)}" )
-      return unless File.file?(path)
-
-      env['sinatra.static_file'] = path
+      public_dir = settings.public_folder
```

Comment:
Why did you move this up here? I don't see why the one-liner is removed.

Reply:
This was done for clarity but shouldn't change functionality. I don't feel strongly about it though; if you feel strongly I'm happy to use the one-liner.

Reply:
@makushline why don't we go back to the one-liner for now.